gui/src/redux/slices/stateSlice.ts

```diff
@@ -270,10 +272,9 @@ export const stateSlice = createSlice({
           : [{ ...item, editing: true }];
         return;
       }
-      // TODO
     },
     addContextItems: (state, action: PayloadAction<ContextItemWithId[]>) => {
-      state.contextItems = state.contextItems.concat(action.payload);
+      state.contextItems.push(...action.payload);
```

Comment:
Not sure if we should use `push` here, I think for state we want things to be immutable, so should return a new object like `concat` does.